modules/caddyhttp/reverseproxy/streaming.go

```diff
@@ -225,7 +226,7 @@ func (h Handler) copyBuffer(dst io.Writer, src io.Reader, buf []byte) (int64, er
 			if rerr == io.EOF {
 				rerr = nil
 			}
-			return written, rerr
+			return written, fmt.Errorf("reading: %v", rerr)
```

Comment:
This error is nil. It shouldn't be wrapped, just returned or nil in place.

Reply:
It's only nil if EOF. I guess we should just change the EOF condition to return instead